core/src/main/java/io/questdb/PropServerConfiguration.java

```diff
@@ -507,6 +508,7 @@ public PropServerConfiguration(
         this.walSegmentRolloverRowCount = getLong(properties, env, PropertyKey.CAIRO_WAL_SEGMENT_ROLLOVER_ROW_COUNT, 200_000);
         this.walWriterDataAppendPageSize = Files.ceilPageSize(getLongSize(properties, env, PropertyKey.CAIRO_WAL_WRITER_DATA_APPEND_PAGE_SIZE, Numbers.SIZE_1MB));
         this.walSquashUncommittedRowsMultiplier = getDouble(properties, env, PropertyKey.CAIRO_WAL_SQUASH_UNCOMMITTED_ROWS_MULTIPLIER, 20.0);
+        this.walMaxLagTxnCount = getInt(properties, env, PropertyKey.CAIRO_WAL_MAX_LAG_TXN_COUNT, Math.max((int) Math.round(walSquashUncommittedRowsMultiplier), 1));
```

Comment:
> Also, the situation you're trying to solve perhaps can be solved by setting

No, I tried that setting and saw the same behavior with unbounded growth of lag txn count.

Reply:
> What's the default in real numbers?

The default is 20.

> I'd suggest it should be very high, in millions so that it does not have any effect by default.

Even 20 transactions is a high default which is hard to reach unless you have many large tables + heavy O3.

> I have doubts that this setting will help in most of the writing patterns.

No, it won't. It's aimed to avoid certain edge cases, nothing more than that.